end_to_end_tests/golden-record/my_test_api_client/api/tests/defaults_tests_defaults_post.py

```diff
@@ -7,67 +7,76 @@
 from ...client import Client
 from ...models.an_enum import AnEnum
 from ...models.http_validation_error import HTTPValidationError
-from ...types import Response
+from ...types import UNSET, Response
 
 
 def _get_kwargs(
     *,
     client: Client,
     json_body: Dict[Any, Any],
-    string_prop: Optional[str] = "the default string",
-    datetime_prop: Optional[datetime.datetime] = isoparse("1010-10-10T00:00:00"),
-    date_prop: Optional[datetime.date] = isoparse("1010-10-10").date(),
-    float_prop: Optional[float] = 3.14,
-    int_prop: Optional[int] = 7,
-    boolean_prop: Optional[bool] = False,
-    list_prop: Optional[List[AnEnum]] = None,
-    union_prop: Optional[Union[Optional[float], Optional[str]]] = "not a float",
-    enum_prop: Optional[AnEnum] = None,
+    string_prop: str = "the default string",
+    datetime_prop: datetime.datetime = isoparse("1010-10-10T00:00:00"),
+    date_prop: datetime.date = isoparse("1010-10-10").date(),
+    float_prop: float = 3.14,
+    int_prop: int = 7,
+    boolean_prop: bool = cast(bool, UNSET),
```

Comment:
I think the right thing to do with these would be to make the type `Union[bool, Unset]` when the prop is not required. That way we're not lying about the data that could be in an instance of this class.